pkg/osd_cluster_upgrader/upgrader_test.go

```diff
@@ -438,36 +424,16 @@ var _ = Describe("ClusterUpgrader", func() {
 				mockKubeClient.EXPECT().Status().AnyTimes().Return(sw)
 				sw.EXPECT().Update(gomock.Any(), gomock.Any()).AnyTimes()
 			})
-			It("Indicates the error in the UpgradeConfig step's history", func() {
-				err := cu.UpgradeCluster(upgradeConfig, logger)
-				stepHistoryReason := upgradeConfig.Status.History.GetHistory(upgradeConfig.Spec.Desired.Version).Conditions.GetCondition(step1).Reason
-				stepHistoryMsg := upgradeConfig.Status.History.GetHistory(upgradeConfig.Spec.Desired.Version).Conditions.GetCondition(step1).Message
+			It("Indicates the error in the condition", func() {
+				_, condition, err := cu.UpgradeCluster(upgradeConfig, logger)
+				stepHistoryReason := condition.Reason
+				stepHistoryMsg := condition.Message
 				Expect(stepHistoryReason).To(Equal(string(step1) + " not done"))
 				Expect(stepHistoryMsg).To(Equal("step " + string(step1) + " failed"))
 				Expect(stepCounter[step1]).To(Equal(1))
-				// TODO - this needs eyes in cluster-upgrader - need to doublecheck whether it is expected to return nil here
```

Comment:
cc @mrbarge This was incorrect. I added a multi error here to report UpgardeCluster error and status update error:
https://github.com/sedroche/managed-upgrade-operator/blob/c8932877cc7335bf5af0dbd9f5bd1d5efff90c12/pkg/controller/upgradeconfig/upgradeconfig_controller.go#L208-L223